screeps-game-api/src/lib.rs

```diff
@@ -4,38 +4,23 @@
 //!
 //! ## `check-all-casts`
 //!
-//! By default, `screeps-game-api` assumes that the Screeps JavaScript API calls return the types
-//! that they are documented to return and bypasses `instanceof` checks when constructing rust
-//! wrappers for those return values.
+//! By default, `screeps-game-api` assumes that the Screeps JavaScript API calls
+//! return the types that they are documented to return and bypasses
+//! `instanceof` checks when constructing rust wrappers for those return values.
 //!
-//! To enable checking all types on all API calls, even ones when the screeps server reliably
-//! returns the expected type, depend on `screeps-game-api` with the `"check-all-casts"` feature
-//! flag:
+//! To enable checking all types on all API calls, even ones when the screeps
+//! server reliably returns the expected type, depend on `screeps-game-api` with
+//! the `"check-all-casts"` feature flag:
 //!
 //! ```toml
 //! [dependencies]
 //! # ...
 //! screeps-game-api = { version = "0.3", features = ["check-all-casts"] }
 //! ```
 #![recursion_limit = "128"]
-#[macro_use]
-extern crate log;
-#[macro_use]
-extern crate num_derive;
-extern crate num_traits;
-#[macro_use]
-extern crate scoped_tls;
-extern crate serde;
-#[macro_use]
-extern crate serde_derive;
-extern crate serde_json;
-#[macro_use]
-extern crate stdweb;
-#[macro_use]
-extern crate stdweb_derive;
 
 #[macro_use]
-mod macros;
+pub(crate) mod macros;
```

Comment:
So, available down the module tree?

Reply:
Yeah - since everything is in `lib.rs` I think it should just work. I'm not 100% sure whether it'd be more idiomatic to use `pub(crate)` or not, but it kind of jumped out of me, as I read `pub` as kind of "this needs to be accessible from a module above this one".